core/typedef.go

```diff
@@ -64,6 +66,10 @@ func (fn Function) Clone() *Function {
 	if fn.ReturnType != nil {
 		cp.ReturnType = fn.ReturnType.Clone()
 	}
+	//this is ugly, but lets try it
+	if fn.ctx != nil {
+		cp.ctx = fn.ctx
+	}
```

Comment:
nit: we can remove this now, `WithArg` should take a `context` arg, as with all the other typedef methods.